packages/hebao_v1/contracts/modules/core/ForwarderModule.sol

```diff
@@ -65,17 +67,21 @@ contract ForwarderModule is BaseModule
         public
         view
     {
+        // Since this contract is a module, we need to prevent wallet from interacting with
+        // Stores via this module. Therefore, we must carefully check the 'to' address as follows,
+        // so no Store can be used as 'to'.
         require(
-            (to == from) ||
-            (to == controller.walletFactory()) ||
-            (to != address(this) && Wallet(from).hasModule(to)),
+            to == controller.walletFactory() || // 'from' can be the wallet to create (not its owner),
+                                                // or an existing wallet,
+                                                // or an EOA iff gasPrice == 0
+            controller.moduleRegistry().isModuleRegistered(to) && (to != address(this)) ||
+            controller.walletRegistry().isWalletRegistered(from) && (to == from),
```

Comment:
>  controller.walletRegistry().isWalletRegistered(from) && (to == from)

We cannot allow this, otherwise the wallet owner can call all critical functions on the wallet (transact, setOwner,...). Should be removed and as far as I can tell that should be fine. The only thing we probably want to do is to allow the wallet owner to call `addModule` on the base wallet so he can't end up with no modules and no wait to add any.

Reply:
I got this fixed.